File: src/ch05/HighOrder.java
package ch05;

import java.util.function.Consumer;

public class HighOrder {
    static void forEach(int[] array, Consumer<Integer> action ) {
        for(int element : array) action.accept(element);
    }

    public static void main(String[] args) {
        forEach(new int[]{1, 2, 3, 4}, new Consumer<Integer>() {
            @Override
            public void accept(Integer it) {
                    if( it < 2 || it > 3) return;
                    System.out.println();
            }
        });
    }
}

File: src/ch05/HighOrder.kt
package ch05


fun forEach(a: IntArray, action: (Int) -> Unit) {
    for(n in a) action(n)
}
fun main() {
    forEach(intArrayOf(1, 2, 3, 4)) {
        if( it < 2 || it > 3 ) return@forEach
    }

    forEach(intArrayOf(1, 2, 3, 4), {it: Int ->
        if( it < 2 || it > 3) return@forEach
    } )
}